// rentpe-ui-component-library/dist/index.modern.js
import React from 'react';
import { Button, Input } from 'antd';
import 'antd/dist/antd.css';

var CustomButton = function CustomButton(props) {
  var children = props.children,
      backgroundColor = props.backgroundColor,
      color = props.color,
      size = props.size,
      height = props.height,
      width = props.width,
      shape = props.shape,
      htmlType = props.htmlType,
      onClick = props.onClick,
      disabled = props.disabled;
  var antdDefaultProps = {};
  if (htmlType) antdDefaultProps = {
    htmlType: htmlType
  };
  var styles = {
    height: '',
    width: '',
    backgroundColor: '',
    color: ''
  };
  if (size) styles.size = size;
  if (color) styles.color = color;
  if (backgroundColor) styles.backgroundColor = backgroundColor;
  if (height) styles.height = height;
  if (width) styles.width = width;
  if (shape) styles.shape = shape;
  return React.createElement(Button, Object.assign({}, antdDefaultProps, {
    disabled: disabled,
    shape: shape,
    size: size,
    style: styles,
    onClick: onClick
  }), children);
};
CustomButton.displayName = 'CustomButton';
CustomButton.defaultProps = {
  children: 'hii joveo',
  size: 'middle',
  shape: 'default'
};

var SearchInput = function SearchInput() {
  var Search = Input.Search;

  var onSearch = function onSearch(value) {
    return console.log(value);
  };

  return React.createElement(Search, {
    placeholder: "input search text",
    onSearch: onSearch,
    style: {
      width: 200
    }
  });
};

export { CustomButton, SearchInput };
